src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,130 @@
+    public Optional<Book> findById(Long id) {
+        String query = "SELECT * FROM books WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            preparedStatement.setLong(1, id);
+
+            ResultSet resultSet = preparedStatement.executeQuery();
+            while (resultSet.next()) {
+                return Optional.of(getResultSet(resultSet));
+            }
+
+        } catch (SQLException ex) {
+            throw new DataProcessingException("Can't find a book by id " + id, ex);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public List<Book> findAll() {
+        List<Book> books = new ArrayList<>();
+        String query = " SELECT * FROM books";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+
+            ResultSet resultSet = preparedStatement.executeQuery();
+            while (resultSet.next()) {
+                books.add(getResultSet(resultSet));
+            }
+
+        } catch (SQLException ex) {
+            throw new DataProcessingException("Cannot create connection to the DB", ex);
+        }
+        return books;
+    }
+
+    @Override
+    public Book update(Book book) {
+        String query = "UPDATE books SET title = ?, price = ? WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+
+            preparedStatement.setString(1, book.getTitle());
+            preparedStatement.setBigDecimal(2, book.getPrice());
+            preparedStatement.setLong(3, book.getId());
+
+            int updatedRows = preparedStatement.executeUpdate();
+            if (updatedRows < 1) {
+                throw new RuntimeException("Can't update DB with book id " + book.getId());
+            }
+            return book;
+
+        } catch (SQLException ex) {
+            throw new DataProcessingException("Can't update database", ex);
+        }
+    }
+
+    @Override
+    public boolean deleteById(Long id) {
+        String query = "DELETE FROM books WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            preparedStatement.setLong(1, id);
+            int deletedRows = preparedStatement.executeUpdate();
+            if (deletedRows < 1) {
+                throw new RuntimeException("Expected to delete one row, but deleted 0 rows");
+            }
+            return deletedRows > 0;
+        } catch (SQLException ex) {
+            throw new DataProcessingException("Can't delete a book by id " + id, ex);
+        }
+    }
+
+    private static Book getResultSet(ResultSet resultSet) throws SQLException {
```

Comment:
bad naming, you are getting here a book from the result set - so name it corectly